File: SEMESTRE_1/CALCUL_SCIENTIFIQUE/bidouille_CAPUTO/cours/8cours/inve.m
%--------------------
%inverse matrice triang <
%
%--------------------
function [ILM]= inve(L,n)
b = zeros(n,1); x = zeros(n,1);ILM=zeros(n,0);
for j=1 :n
   b(:)=0; b(j)=1;
   for i=1 :n
     x(i) = b(i)- L(i,1:i-1)*x(1:i-1);
   end
   ILM = [ILM x];
end

File: SEMESTRE_1/CALCUL_SCIENTIFIQUE/bidouille_CAPUTO/calcul/tableaux_dyn/inve.f
c inversion matrice triang <
c entrees n ,l, 
c sortie il

      subroutine  inve(n,l,il,c,y)
      implicit none
      integer n,i,j
      double precision, dimension(n,n) :: l,il
      double precision, dimension(n) :: y,c

      do j=1,n
        c=0.
        c(j)=1.
        do i=1,n
         y(i)=(c(i)-sum(l(i,1:i-1)*y(1:i-1)))/l(i,i)
        enddo
        il(:,j)=y
      enddo
      end

